File: c++/Assignment_1.cpp
#include <iostream>
using namespace std;


int main()
{
    //input of total number of students and their marks//
    int n;
    
    cout<<"enter number of students:-";
    cin>>n;
    int marks[n];
    for(int i=0;i<n;i++){
        cout<<"enter marks of students (if absent enter -1):-";
        cin>>marks[i];
    }


    // averge score of class
    int sum=0;
    float average;
    int absent=0;
    for(int j=0;j<n;j++)
    {
        if(marks[j]!=-1)
        {
            sum=sum+marks[j];
        }
        else{
            absent=absent+1;
        }
    }
    average=sum/n;
    cout<<"the average score of class is:-  "<<average<<endl;

    //highest score in the class
    int temp=marks[0];
    for(int k=1;k<n;k++)
    {
        if(temp<marks[k]){
            if(marks[k]!=-1)
        {
            temp=marks[k];
        }
            
        }
    }
    cout<<"the highest score is:-  "<<temp<<endl;
    

    //lowest score in class
    int low=marks[0];
    for(int h=1;h<n;h++){
        if(low>marks[h]){
            if(marks[h]!=-1)
        {
            low=marks[h];
        }
            
        }
    }
    cout<<"the lowest score in the class is:-  "<<low<<endl;


    //count of absent student
    cout<<"The count of absent student is:-  "<<absent<<endl;


    //marks with highest frequency
    



    return 0;
}

File: python/assignment 1.py
studentmarks=[]
def getNoStudent():
    print("enter the numberof students")
    nstud=int(input())
    for i in range(nstud):
        print("enter marks of student")
        marks=int(input())
        studentmarks.append(marks)
        print(marks)

def averege():
    sum=0
    num=0
    for i in range(len(studentmarks)):
        if studentmarks[i] !=-22:
            sum=sum+studentmarks[i]
            num+=1
        print("total marks",sum)
        average=sum/num
        print(average) 
        print("Highhest marks" ,max(studentmarks))
        print("minimum marks",min(student))
def absentStudent():
    absent=0
    for i in range(len(studentmarks)):
        if i==-22:
            absent=absent+1
getNoStudent()
averege()
